### ISSUE TITLE
Hooks that have `stages:` set in .pre-commit-hooks.yaml do not respect caller's `default_stages` setting

### ISSUE THREAD

**Issue by rpdelaney** [pre-commit/pre-commit-hooks#515]
I just added my first pre-push hook as an experiment. For now, I want to run that hook when pushing, but not all the others. So I set `default_stages: [commit]` in my [`.pre-commit-config.yaml`](https://gist.github.com/rpdelaney/1a5b366f9040e6ee81424df4783ae341).

Here's what I see on push:

```console
$ git push
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
Check that executables have shebangs.................(no files to check)Skipped
run all tests............................................................Passed
```
I expected only that last one, but instead I got the first three as well.

In this repository it looks like these are the only hooks that have their own `stages` setting. I'd venture a guess that that setting is somehow over-riding the `default_hooks` setting in a caller's config.

**Comment by asottile** [comment]
yep this is [working as intended](https://pre-commit.com/#top_level-default_stages)

> (optional: default (all stages)) a configuration-wide default for the stages property of hooks. This will only override individual hooks that do not set stages.

**Comment by rpdelaney** [comment]
Hrm. Is there another way I could prevent these hooks from running on push?

**Comment by asottile** [comment]
override their `stages` property

**Comment by rpdelaney** [comment]
Ok. Just to be clear, I don't think there's any problem with how `pre-commit` is handling this -- but it is a bit of a surprise that when adding hooks from this repo, these three of them are unaffected by the `default_stages` setting and require an explicit over-ride, while the others don't. Not any kind of big deal though, of course. Thanks

**Comment by asottile** [comment]
the reason for that if you dig in: pre-commit/pre-commit-hooks#315 pre-commit/pre-commit-hooks#361
